### ISSUE TITLE
Monaco editor syntax highlights only after scroll

### ISSUE THREAD

**Issue by mac501pl** [microsoft/monaco-editor#2187]
**monaco-editor version:** [react-monaco-editor](https://github.com/react-monaco-editor/react-monaco-editor)
**Browser: chromium (electron)**
**OS: Windows 10**

I'm using react wrapper for Monaco Editor. I have already filed an [issue](https://github.com/react-monaco-editor/react-monaco-editor/issues/312) on their page but I was redirected here.

Syntax highlighter for custom language only fires after i scroll through text.

![recording](https://user-images.githubusercontent.com/42701371/95677440-df6f2c80-0bc5-11eb-9c74-2196f81873ba.gif)
[repository](https://github.com/mac501pl/8BitSimulator)

The linked issue also includes minimal reproduction steps

**Comment by mac501pl** [comment]
I've just chacked with vanilla monaco-editor 0.21.2. The issue still exists. Code:
```ts
import * as monaco from "monaco-editor";
import * as React from "react";
import { ReactElement, ReactPropTypes } from "react";
import Intel8080LanguageDefinition from "@main/editor/Intel8080LanguageDefinition";
import * as fs from "fs";

export default class Editor extends React.Component {
	constructor(props: ReactPropTypes) {
		super(props);
	}

	componentDidMount(): void {
		monaco.languages.register({ id: "8080asm" });
		monaco.languages.setMonarchTokensProvider("8080asm", new Intel8080LanguageDefinition().getDefinition());
		monaco.editor.defineTheme("8080-dark", {
			base: "vs-dark",
			inherit: true,
			rules: [
				{ token: "other", foreground: "ffffff" },
				{ token: "mnemonic", foreground: "CB772F", fontStyle: "bold" },
				{ token: "number", foreground: "6897BB" },
				{ token: "register", foreground: "FFC66D" },
				{ token: "declaration", foreground: "BED6FF" },
				{ token: "comment ", foreground: "619647", fontStyle: "italic" },
				{ token: "label", foreground: "9876AA" },
				{ token: "quotes", foreground: "A5C25C" }
			],
			colors: {
				"editor.background": "#2B2B2B"
			}
		});
		monaco.editor.create(document.getElementById("editor"), {
			value:fs.readFileSync(
				"./assets/test_files/public/test.txt",
				"utf8"
			),
			theme: "8080-dark",
			language: "8080asm",
			smoothScrolling: true,
			mouseWheelZoom: true,
			minimap: {
				enabled: false
			},
			fontFamily: "Consolas",
			cursorSmoothCaretAnimation: true,
			cursorBlinking: "smooth",
			automaticLayout: true
		});
	}

	getEditorValue(): string {
		return monaco.editor.getModels()[0].getValue();
	}

	render(): ReactElement {
		return (<div style={{height: "100vh", width: "50%"}} id="editor"/>);
	}
```

**Comment by zerooverture** [comment]
I have also encountered the same problem, which has not been solved at present.

**Comment by Prinzhorn** [comment]
Can you create a reproducible example that can be run locally? Without react? Preferably based on one of the [official examples](https://github.com/microsoft/monaco-editor-samples/)? The repository you've linked is 404 and the code you've posted requires build steps and files you did not include.

**Comment by andyboyne** [comment]
I am hitting the exact same problem. No syntax highlight until after scroll. @Prinzhorn  did you find a workaround?

**Comment by mac501pl** [comment]
Sorry for my lack of input for an extended time, I was preoccupied and unfortunately I don't have time to try and make a reproducable example. Let me give you guys some more details though.

I was indeed unable to reproduce the problem with the playground. I suspect this might be a Webpack or Electron related issue.

I actually don't experience this problem anymore in my application, it apparently fixed itself on its own, probably when I was upgrading packages. Let me explain everything in more detail.

I was using Electron 8, React, [react-monaco-editor](https://github.com/react-monaco-editor/react-monaco-editor) and Webpack 4. After I upgraded to Electron 10 and Webpack 5 the highlighter works fine.

What's more, after rebundling the code and rerunning the electron application the syntax highlighter worked fine (worked before scrolling). Only after refreshing the page (with a regular `Ctrl + R`), the highlighter would break and would look like on the example.

**Comment by andyboyne** [comment]
So you still have an issue with Ctrl+R? Upon further inspection this is exactly what I see. Initial load fine, subsequent reloads broken until scroll.  

Interestingly, I also am using webpack 5 and electron 10.

**Comment by mac501pl** [comment]
No, I don't have any issues anymore. All I did was bulk upgrading all the packages I used.

**Comment by andyboyne** [comment]
After tracing through the monaco code I narrowed it down to a platform.setImmediate(()=> not firing after reload, which then led me to: https://github.com/electron/electron/issues/25314.

Tl;dr; You must be on electron 10.0.4+ for monaco to work after reload.
